### PR TITLE
feat(BTable): custom filter functions

### PR DESCRIPTION
**Pull request by kevinkosterr**
# Describe the PR
This PR aims to bring over the ability to pass a custom filter function to the `BTable` component as seen in `bootstrap-vue`. This gives developers more control over how exactly they want to filter their data inside tables.

## Small replication
Hard to properly convey in a video, but here is a small replication of what this could do. Code is displayed underneath the table.

https://github.com/user-attachments/assets/232fbd06-ed5b-4087-9f37-17081af03b2c

## PR checklist

**What kind of change does this PR introduce?** (check at least one)

- [ ] Bugfix :bug: - `fix(...)`
- [x] Feature - `feat(...)`
- [ ] ARIA accessibility - `fix(...)`
- [x] Documentation update - `docs(...)`
- [ ] Other (please describe)

**The PR fulfills these requirements:**

- [x] Pull request title and all commits follow the [**Conventional Commits**](https://www.conventionalcommits.org/) convention or has an [**override**](https://github.com/googleapis/release-please#how-can-i-fix-release-notes) in this pull request body **This is very important, as the `CHANGELOG` is generated from these messages, and determines the next version type. Pull requests that do not follow conventional commits or do not have an override will be denied**

### CODE REVIEW

File: packages/bootstrap-vue-next/src/components/BTable/BTable.vue [hunk 1]
```diff
@@ -658,6 +659,11 @@ const computedItems = computed<Items[]>(() => {
               (!props.filterable?.includes(key) && !!props.filterable?.length)
             )
               return false
+
+            if (props.filterFunction && typeof props.filterFunction === 'function') {
+              return props.filterFunction(item)
```
Comment (VividLemon) [inline packages/bootstrap-vue-next/src/components/BTable/BTable.vue:664]: The second param according to the bootstrap-vue docs is `the content of the filter prop (could be a string, RegExp, array, or object)`, so `(item, prop.filter)`

**Review by VividLemon — CHANGES_REQUESTED** [review]

File: packages/bootstrap-vue-next/src/components/BTable/BTable.vue (hunk 1 above)
Comment (VividLemon) [reply packages/bootstrap-vue-next/src/components/BTable/BTable.vue:664]: Idrk why it exists this way

File: packages/bootstrap-vue-next/src/types/ComponentProps.ts [hunk 2]
```diff
@@ -1092,10 +1092,10 @@ export interface BTableProps<Items> extends Omit<BTableLiteProps<Items>, 'tableC
   perPage?: Numberish
   currentPage?: Numberish
   filter?: string
+  filterFunction?: (item: Items) => boolean
```
Comment (VividLemon) [inline packages/bootstrap-vue-next/src/types/ComponentProps.ts:1095]: I think this may be `(item: Readonly<Item>, filter: string | undefined) => boolean`

**Comment by VividLemon** [comment]
I'm also not quite sure how I feel about this prop in general. Why not do `filter?: string | ((item: Readonly<Item>) => boolean)`? Why need the extra prop?

**Comment by kevinkosterr** [comment]
@VividLemon I'm not sure why we'd use the `filter` prop for this, personally it seems more readable and clear being its own prop. I feel it would add more unnecessary complexity to cram it in with the existing `filter` prop.

**Pull request merged**